sdk/core/core/inc/az_static_assert.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STATIC_ASSERT_H
+#define AZ_STATIC_ASSERT_H
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AZ_STATIC_ASSERT(CONDITION) typedef int _az_static_assert[(CONDITION) ? 1 : -1];
```

Comment:
suggestion: no action required

Maybe do _Static_assert if the compiler supports it, this gives a way better error message.